File: palindrome/palindrome.c
#include <stdio.h> 
int main()
{
   int n, rev=0,n1;
 
   printf("Enter a number to reverse\n");
   scanf("%d",&n);
    n1=n;
 
   while (n != 0)
   {
      rev=rev*10;
      rev=rev+n%10;
      n = n/10;
   }
    if(n1==rev)
   printf("\nthe given number is a palindrome\n");
  else
 printf("not palindrome");
 
   return 0;
}

File: palindrome/palindrome.rb
rev=0
pal=0
puts "enter any number"
n=gets.to_i
pal=n
while n > 0
      r = n % 10
      rev = ( rev * 10 ) + r
      n=n/10
end
if pal==rev
 puts "given number is a palindrome"
 else
 puts "not palindrome"
end
